irods_consortium_continuous_integration_build_hook.py

```diff
@@ -9,65 +9,49 @@
-        elif irods_python_ci_utilities.get_distribution_version_major() == '14':
-            irods_python_ci_utilities.make_symbolic_link_as_root('/usr/lib/x86_64-linux-gnu/libglobus_callout.so.0', '/usr/lib/x86_64-linux-gnu/libglobus_callout.so')
-            irods_python_ci_utilities.make_symbolic_link_as_root('/usr/lib/x86_64-linux-gnu/libglobus_gss_assist.so.3', '/usr/lib/x86_64-linux-gnu/libglobus_gss_assist.so')
-        elif irods_python_ci_utilities.get_distribution_version_major() == '16':
-            irods_python_ci_utilities.make_symbolic_link_as_root('/usr/lib/x86_64-linux-gnu/libglobus_callout.so.0', '/usr/lib/x86_64-linux-gnu/libglobus_callout.so')
-            irods_python_ci_utilities.make_symbolic_link_as_root('/usr/lib/x86_64-linux-gnu/libglobus_gss_assist.so.3', '/usr/lib/x86_64-linux-gnu/libglobus_gss_assist.so')
-
-def install_build_prerequisites_yum():
-    irods_python_ci_utilities.install_os_packages(get_build_prerequisites())
-    irods_python_ci_utilities.make_symbolic_link_as_root('/usr/lib64/libglobus_callout.so.0', '/usr/lib64/libglobus_callout.so')
+def add_cmake_to_front_of_path():
+    cmake_path = '/opt/irods-externals/cmake3.11.4-0/bin'
+    os.environ['PATH'] = os.pathsep.join([cmake_path, os.environ['PATH']])
+
+def install_building_dependencies(externals_directory):
+    externals_list = ['irods-externals-cmake3.11.4-0',
+                      'irods-externals-clang6.0-0',
+                      'irods-externals-cppzmq4.2.3-0',
+                      'irods-externals-libarchive3.3.2-1',
+                      'irods-externals-avro1.9.0-0',
+                      'irods-externals-clang-runtime6.0-0',
+                      'irods-externals-boost1.67.0-0',
+                      'irods-externals-json3.7.3-0',
+                      'irods-externals-zeromq4-14.1.6-0']
+    if externals_directory == 'None' or externals_directory is None:
+        irods_python_ci_utilities.install_irods_core_dev_repository()
+        irods_python_ci_utilities.install_os_packages(externals_list)
+    else:
+        package_suffix = irods_python_ci_utilities.get_package_suffix()
+        os_specific_directory = irods_python_ci_utilities.append_os_specific_directory(externals_directory)
+        externals = []
+        for irods_externals in externals_list:
+            externals.append(glob.glob(os.path.join(os_specific_directory, irods_externals + '*.{0}'.format(package_suffix)))[0])
+        irods_python_ci_utilities.install_os_packages_from_files(externals)
+    add_cmake_to_front_of_path()
+    install_os_specific_dependencies()
+
+def install_os_specific_dependencies_apt():
+    irods_python_ci_utilities.install_os_packages(['globus-proxy-utils', 'libglobus-gssapi-gsi-dev', 'make', 'libssl-dev', 'libkrb5-dev', 'gcc'])
+    if irods_python_ci_utilities.get_distribution_version_major() == '16':
```

Comment:
so this works fine on ub18?

Reply:
It builds on ub18, but tests don't pass on ubuntu 18. The tests pass on ubuntu 16 and centOS 7. it fails during setup when the command ```grid-cert-request -nopw -force -cn gsi_client_user``` is run. 
```
stderr: Can't load /var/adm/wtmp into RNG
140420625727936:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/var/adm/wtmp
Can't load /var/log/messages into RNG
140420625727936:error:2406F079:random number generator:RAND_load_file:Cannot open file:../crypto/rand/randfile.c:88:Filename=/var/log/messages
Error number 1 was returned by 
   /usr/bin/openssl
```